### PR TITLE
MM-61043 Error deleting board if fileID is empty

### PR DESCRIPTION
**Pull request by Rajat-Dabade**
#### Summary

https://hub.mattermost.com/icu/pl/qqb9tqrzdtfajyfrysd39ecaro

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-61043

### CODE REVIEW

**Review by harshilsharma63 — APPROVED** [review]

**Comment by harshilsharma63** [comment]
@Rajat-Dabade can you add the steps to reproduce this bug in the ticket?

File: server/app/files.go [hunk 1]
```diff
@@ -60,7 +60,16 @@ func (a *App) GetFileInfo(filename string) (*mm_model.FileInfo, error) {
...
-	fileInfoID := getFileInfoID(strings.Split(filename, ".")[0])
+	parts := strings.Split(filename, ".")
+	if len(parts) < 1 || len(parts[0]) <= 1 {
+		return nil, model.NewErrNotFound("No file found with name" + filename)
+	}
+
+	fileInfoID := getFileInfoID(parts[0])
+
+	if (fileInfoID) == "" {
+		return nil, nil
```
Comment (hmhealey) [inline server/app/files.go:71]: Should this case also return an error? I'd expect that this sort of method would always return either a FileInfo or an error, but this case returns nil for both of them

**Review by hmhealey — APPROVED** [review]
Looks good with one caveat

**Comment by hmhealey** [comment]
Also, excellent explanation of your investigation here. I'm surprised that we keep the file extension on the uploaded file and that we have to calculate the file ID from the filename, so it's good to know that the code works that way

File: server/app/files.go (hunk 1 above)
Comment (Rajat-Dabade) [reply server/app/files.go]: Done, returning an error with message `file not found`.

**Pull request merged**